skbio/io/_registry.py

```diff
@@ -485,7 +471,7 @@ def _rw_getter(name, fmt, cls):
     return None
 
 
-def sniff(fp, cls=None, mode='U'):
+def sniff(fp, cls=None, mode='r'):
```

Comment:
mode still shouldn't be user facing, but now we have the potential for sniffers to disagree about what mode the file should be. This is problematic, I think there may be ways around this entirely though. (looking into them)